pkg/zk/generators.go

```diff
@@ -247,7 +247,8 @@ func MakeHeadlessService(z *v1beta1.ZookeeperCluster) *v1.Service {
 
 func makeZkConfigString(z *v1beta1.ZookeeperCluster) string {
 	ports := z.ZookeeperPorts()
-	return "4lw.commands.whitelist=cons, envi, conf, crst, srvr, stat, mntr, ruok\n" +
+
+	zkConfig := "4lw.commands.whitelist=cons, envi, conf, crst, srvr, stat, mntr, ruok\n" +
 		"dataDir=/data\n" +
 		"standaloneEnabled=false\n" +
 		"reconfigEnabled=true\n" +
```

Comment:
We  have to ensure that  these values are not overridden by the user,
```"dataDir=/data\n" +
		"standaloneEnabled=false\n" +
		"reconfigEnabled=true\n" +
                 "SkipACL=yes \n"
```
If we move these fields below, will that serve the purpose?

Reply:
good catch @anishakj !

Moved the optional parameters at the top of the file.
This way - even if they declare one of the reserved params, this being a java properties file - last entry wins